openstack/baremetal/v1/nodes/doc.go

```diff
@@ -4,10 +4,24 @@ package nodes
 Package nodes provides information and interaction with the nodes API
 resource in the OpenStack Bare Metal service.
 
+	// Example to List Nodes with Detail
+	nodes.ListDetail(client, nodes.ListOpts{}).EachPage(func(page pagination.Page) (bool, error) {
+		nodeList, err := nodes.ExtractNodes(page)
+		if err != nil {
+			return false, err
+		}
+
+		for _, n := range nodeList {
+			// Do something
+		}
+
+		return true, nil
+	})
+
 	// Example to List Nodes
 	nodes.List(client, nodes.ListOpts{
-		Detail: true,
 		ProvisionState: Deploying,
+		Fields: []string{"name"},
```

Comment:
Is this going to work? I think you treat UUID as required.

Reply:
What do you mean? ListOpts has no required fields.

Reply:
I'm just asking if your code is ready to handle nodes with UUID missing. I thought no, but now I've skimmed through the code and it seems okay to me.

Reply:
The `Node` struct is pretty happy to deserialize everything I've seen the Ironic API throw at it.